lifecycle.Rmd

```diff
@@ -7,52 +7,251 @@ status("drafting")
+## Major vs. minor vs. patch release
+
+The version number of your package is always increasing, but it's more than just an incrementing counter -- the way the number changes with each release can convey information about the nature of the changes.
+The transition from 0.3.1 to 0.3.2, which is a patch release, has a very different vibe from the transition from 0.3.2 to 1.0.0, which is a major release.
+A package version number can also convey information about where the package is in its lifecycle.
+For example, the version 1.0.0 often signals that the public interface of a package is considered stable.
+
+How do you decide which type of release to make, i.e. which component(s) of the version should you increment?
+A key concept is whether the associated changes are backward compatible, meaning that pre-existing code will still "work" with the new version.
+We put "work" in quotes, because this designation is open to a certain amount of interpretation.
+A hardliner might take this to mean "the code works in exactly the same way, in all contexts, for all inputs".
+A more pragmatic interpretation is that "the code still works, but could produce a different result in some edge cases".
+A change that is not backward compatible is often described as a *breaking* change.
+In section @sec-lifecycle-backward-compatibility, we'll revisit the topic of backward compatibility.
+Similar to the issue of package dependencies (@sec-dependencies-pros-cons), it's a question of trade-offs.
+
+In practice, backward compatibility is not a clear-cut distinction.
+It is typical to assess the impact of a change from a few angles:
+
+-   Degree of change in behaviour.
+    The most extreme is to make something that used to be possible into an error, i.e. impossible.
+    In contrast, changes to a print method are much less significant.
+
+-   How the changes fit into the design of the package.
+    A change to low-level infrastructure, such as a utility that gets called in all user-facing functions, is more fraught than a change that only affects one parameter of a single function.
+
+-   How much existing usage is affected.
+    This is a combination of how many of your users will perceive the change and how many existing users there are to begin with.
+
+Recall that a version number will have one of these forms, if you're following our conventions:
+
+``` default
+<major>.<minor>.<patch>        # released version
+<major>.<minor>.<patch>.<dev>  # in-development version
+```
+
+If the current package version is `0.8.1.9000`, here's our advice on how to pick the version number for the next release:
+
+-   Increment `patch`, e.g. `0.8.2` for a **patch release**: you've fixed bugs, but you haven't added any significant new features.
+    For example, if we discover a show-stopping bug shortly after a release, we would make a quick patch release with the fix.
+    Most releases will have a patch number of 0.
```

Comment:
Maybe say we don't do blog posts about patch releases because we don't think people know about it?

Reply:
Incorporated